handler/src/main/java/io/netty/handler/ssl/ReferenceCountedOpenSslEngine.java

```diff
@@ -235,37 +235,39 @@ protected void deallocate() {
         } finally {
             readerLock.unlock();
         }
-        ssl = SSL.newSSL(context.ctx, !context.isClient());
-        try {
-            networkBIO = SSL.bioNewByteBuffer(ssl, context.getBioNonApplicationBufferSize());
+        synchronized (this) {
+            ssl = SSL.newSSL(context.ctx, !context.isClient());
```

Comment:
Yeah, create a temporary final local variable outside of the sync this block but inside the reader lock then assign it to ssl in the sync this block so access to the ssl variable is thread safe.